rpcs3/Emu/Cell/lv2/sys_fs.h

```diff
@@ -135,29 +143,25 @@ struct lv2_fs_object
 	// File Name (max 1055)
 	const std::array<char, 0x420> name;
 
-	lv2_fs_object(lv2_fs_mount_point* mp, const char* filename)
+	lv2_fs_object(lv2_fs_mount_point* mp, std::string_view filename)
 		: mp(mp)
 		, name(get_name(filename))
 	{
 	}
 
-	static lv2_fs_mount_point* get_mp(const char* filename);
+	static lv2_fs_mount_point* get_mp(std::string_view filename);
 
-	static std::array<char, 0x420> get_name(const char* filename)
+	static std::array<char, 0x420> get_name(std::string_view filename)
 	{
 		std::array<char, 0x420> name;
 
-		for (auto& c : name)
+		if (filename.size() >= 0x420)
 		{
-			c = *filename++;
-
-			if (!c)
-			{
-				return name;
-			}
+			filename = filename.substr(0, 0x420 - 1);
```

Comment:
```suggestion
			const size_t len = std::min<size_t>(filename.size(), name.size() - 1);
```

With that the only place with `0x420` magic will be in `name` variable declaration.